### PR TITLE
multiple spelling corrections

### PR DESCRIPTION
**Pull request by onlyjob**

### CODE REVIEW

File: CRM/Contact/BAO/GroupNestingCache.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ static public function update() {
     }
 
     if (self::checkCyclicGraph($tree)) {
-      CRM_Core_Error::fatal(ts('We detected a cycle which we cant handle. aborting'));
+      CRM_Core_Error::fatal(ts('We detected a cycle which we can't handle. aborting'));
```
Comment (colemanw) [inline CRM/Contact/BAO/GroupNestingCache.php]: You'll need to surround the string with double quotes or this single quote breaks it.

File: CRM/Core/Controller.php [hunk 2]
```diff
@@ -855,7 +855,7 @@ public function invalidKey() {
...
+    $msg = ts('We can\'t load the requested web page. This page requires cookies to be enabled in your browser settings. Please check this setting and enable cookies (if they are not enabled). Then try again. If this error persists, contact the site administrator for assistance.') . '<br /><br />' . ts('Site Administrators: This error may indicate that users are accessing this page using a domain or URL other than the configured Base URL. EXAMPLE: Base URL is http://example.org, but some users are accessing the page via http://www.example.org or a domain alias like http://myotherexample.org.') . '<br /><br />' . ts('Error type: Could not find a valid session key.');
```
Comment (colemanw) [inline CRM/Core/Controller.php]: Would be great to also fix the escaped single quote in here. Escaped quotes in ts() strings confuses translators, and surrounding the string with double-quotes would fix it.

File: CRM/Core/DAO.php [hunk 3]
```diff
@@ -220,7 +220,7 @@ protected function assignTestValue($fieldName, &$fieldDef, $counter) {
           break;
 
         case CRM_Utils_Type::T_TIME:
-          CRM_Core_Error::fatal('T_TIME shouldnt be used.');
+          CRM_Core_Error::fatal('T_TIME shouldn't be used.');
```
Comment (colemanw) [inline CRM/Core/DAO.php]: Another broken string due to single quote.

File: CRM/Group/Form/Edit.php [hunk 4]
```diff
@@ -84,7 +84,7 @@ class CRM_Group_Form_Edit extends CRM_Core_Form {
    * Set up variables to build the form.
    *
    * @return void
-   * @acess protected
+   * @access protected
```
Comment (colemanw) [inline CRM/Group/Form/Edit.php]: This can be simply removed. @access is an old php4 annotation that we're no longer using.

File: CRM/Report/Form/Contribute/Sybunt.php [hunk 5]
```diff
@@ -514,17 +514,17 @@ public function buildChart(&$rows) {
     $previous_year = $current_year - 1;
     $previous_two_year = $current_year - 2;
     $previous_three_year = $current_year - 3;
-    $upto = $current_year - 4;
+    $up to = $current_year - 4;
```
Comment (colemanw) [inline CRM/Report/Form/Contribute/Sybunt.php]: Variable names can't contain spaces. Let's revert the changes in this file.

File: CRM/Utils/Rule.php [hunk 6]
```diff
@@ -510,7 +510,7 @@ public static function string($value, $maxLength = 0) {
    */
   public static function boolean($value) {
     return preg_match(
-      '/(^(1|0)$)|(^(Y(es)?|N(o)?)$)|(^(T(rue)?|F(alse)?)$)/i', $value
+      '/(^(1|0)$)|(^(Y(es)?|N(o)?)$)|(^(T(rue)?|F(else)?)$)/i', $value
```
Comment (colemanw) [inline CRM/Utils/Rule.php]: I don't think this one was a misspelling. It's part of a regex string.

File: CRM/Utils/String.php [hunk 7]
```diff
@@ -413,7 +413,7 @@ public static function strtoboolstr($str) {
     if (preg_match('/^(y(es)?|t(rue)?|1)$/i', $str)) {
       return '1';
     }
-    elseif (preg_match('/^(n(o)?|f(alse)?|0)$/i', $str)) {
+    elseif (preg_match('/^(n(o)?|f(else)?|0)$/i', $str)) {
```
Comment (colemanw) [inline CRM/Utils/String.php]: Ditto.

File: xml/templates/civicrm_state_province.tpl [hunk 8]
```diff
@@ -3619,7 +3619,7 @@ INSERT INTO civicrm_state_province (id, country_id, abbreviation, name) VALUES
 (5166, 1075, "TR", "Tampere Region"),
 (5167, 1075, "OB", "Ostrobothnia"),
 (5168, 1075, "NK", "North Karelia"),
-(5169, 1075, "NO", "Nothern Ostrobothnia"),
+(5169, 1075, "NO", "Northern Ostrobothnia"),
```
Comment (colemanw) [inline xml/templates/civicrm_state_province.tpl]: I think we'll need to add an upgrade script for this.

**Comment by eileenmcnaughton** [comment]
I love the fact you have done this patch! Looks like there are few complexities in it - you might choose just to remove those parts in order to get the bulk through more easily - ie. don't feel it's on you to deal with the upgrade script item - just remove it from your patch & potentially add a ticket in JIRA.

I'd hate to see this get too hard over the 10% that is a bit tricky

**Comment by colemanw** [comment]
Agreed. I'm happy to help out too, just let me know what of the above you want to fix yourself and what you want help with.

**Comment by onlyjob** [comment]
Thanks for review and apologies for my sloppy patch. I reckon I was tired as I've spotted (and corrected) only some issues but missed so many... Well spotted, @colemanw, hopefully everything should be fixed in 9251aff, please review.

File: xml/templates/civicrm_state_province.tpl (hunk 8 above)
Comment (onlyjob) [inline xml/templates/civicrm_state_province.tpl]: It might be a good idea but it would be a separate task to do... Or is it a blocker to merge?

**Comment by onlyjob** [comment]
Thanks, Eileen. I've made corrections but don't feel qualified to take care of upgrade script and frankly I have no time for this yet. Also not sure how to describe the problem in ticket... Could someone handle an upgrade script (or a ticket) please? Cheers.

**Comment by eileenmcnaughton** [comment]
I've added a ticket for the state on upgrade - https://issues.civicrm.org/jira/browse/CRM-16447.
jenkins, ok to test

**Comment by onlyjob** [comment]
Thank you, @eileenmcnaughton. :)

**Comment by eileenmcnaughton** [comment]
I'm still trying to figure out the command to get jenkins going on this - will try test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by colemanw** [comment]
jenkins, add to whitelist

File: CRM/Contact/BAO/GroupNestingCache.php [hunk 9]
```diff
@@ -69,7 +69,7 @@ static public function update() {
     }
 
     if (self::checkCyclicGraph($tree)) {
-      CRM_Core_Error::fatal(ts('We detected a cycle which we cant handle. aborting'));
+      CRM_Core_Error::fatal(ts("We detected a cycle which we can't handle. aborting"));
```
Comment (colemanw) [inline CRM/Contact/BAO/GroupNestingCache.php]: Hey @mlutfy maybe we should just remove ts() from here. I don't even know what this string means in English!

Comment (mlutfy) [inline CRM/Contact/BAO/GroupNestingCache.php]: @colemanw: then people who aren't fluent in English will be even more frustrated. Why not fix the English string to something meaningful?

Comment (eileenmcnaughton) [inline CRM/Contact/BAO/GroupNestingCache.php]: We could change it to 'please update this error message once you figure out what it means' :-)

Comment (colemanw) [inline CRM/Contact/BAO/GroupNestingCache.php]: Fair enough, I had just remembered @mlutfy mentioned a while ago that low level error messages are rarely seen and hard to google if translated, so best left in English.

Comment (mlutfy) [inline CRM/Contact/BAO/GroupNestingCache.php]: Ah, good point. From loooking at the code, it seems like this is an error that can only happen if an admin makes the mistake of importing groups who have a parent group that can also be a child group, creating a cycle. Very unlikely to be a user error. Although that is a rather subjective, and you can expect me to flip-flop on that issue. I guess these days I'm more in a "all messages should be clear and translated" state of mind. However, not a big deal imho for this error message (I googled it, and besides code references, didn't see a forum post about it).

**Comment by eileenmcnaughton** [comment]
Can I merge? Can I? Can I?

**Comment by colemanw** [comment]
Be my guest @eileenmcnaughton :)

**Comment by eileenmcnaughton** [comment]
Our thpelling is much improved now :-)
